projects/fw_103_hw/src/adc_converter.cpp

```diff
@@ -0,0 +1,58 @@
+#include "adc.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "wait.h"
+#include "log.h"
+
+static void prv_button_press(const GpioAddress *address, void *context) {
+    uint16_t adc_data = 0;
+    adc_read_converted(*read_channel, &adc_data);
+    LOG_DEBUG("%d\n", adc_data);
```

Comment:
I think you forgot a line or two declaring `read_channel`, this won't compile otherwise since `read_channel` isn't visible from `prv_button_press`, it's a local variable in `main`.